cmd/system-probe/api/module/loader.go

```diff
@@ -45,7 +46,8 @@ type loader struct {
 // Register a set of modules, which involves:
 // * Initialization using the provided Factory;
 // * Registering the HTTP endpoints of each module;
-func Register(cfg *config.Config, httpMux *mux.Router, factories []Factory) error {
+// * Register the gRPC server;
+func Register(cfg *config.Config, httpMux *mux.Router, grpcServer *grpc.Server, factories []Factory) error {
```

Comment:
consider refactoring a bit to move the endpoint registration into a separate function (2 functions: http and grpc) and call the relevant function based on the cfg flag.

This to avoid passing the pointer parameter to the function and handling the nil case (when gRPC is disabled)